configGlobal.go

```diff
@@ -125,17 +125,76 @@ func setGlobalConfigVariables(storageDir string, cfgFile string) error {
 		globalLogFileCount = configInt
 	}
 
-	// No form of defaults for E-Mail settings, just read them in
-	emailFromAddress = viper.GetString("emailFromAddress")
-	emailToAddress = viper.GetString("emailToAddress")
-	emailServerHostname = viper.GetString("emailServerHostname")
-	emailServerPort = viper.GetInt("emailServerPort")
-	emailAuthUsername = viper.GetString("emailAuthUsername")
-	emailAuthPassword = viper.GetString("emailAuthPassword")
+	// Support for deprecated -m flag until it gets removed
+	if sendMail {
+		defaultNotifier, err := NewEmailNotifier()
+		if err != nil {
+			return err
+		}
+		onFailureNotifiers = append(onFailureNotifiers, defaultNotifier)
+		onStartNotifiers = append(onStartNotifiers, defaultNotifier)
+		return nil
+	}
+
+	// Try to set default notifiers if non are defined in the config
+	if viper.IsSet("notifications") == false {
```

Comment:
Ok, let me check if I understand: This should be changed to:
if `-m` Flag is set => override notifications settings
else => use notifications settings (with no defaults)

Reply:
I think I was a little crazy when I originally suggested this. Based on my original suggestion, you would never be able to disable email notification (since it would always be defaulted). So, with that in mind:

* If the deprecated `-m` is specified, create a default email notification (which you are doing now)
* If `-m` is not specified, then email should only be enabled if explicitly specified in the global configuration
* If a notification section exists, use it as is (unless `-m` is specified). If email is not in the notification list, then you don't get email.

I think that's what the code is doing today, right?